src/bin/pg_autoctl/pgsql.c

```diff
@@ -1608,6 +1608,85 @@ pgsql_set_default_transaction_mode_read_write(PGSQL *pgsql)
 }
 
 
+/*
+ * pgsql_set_password sets the password for the given userName.
+ */
+bool
+pgsql_set_password(PGSQL *pgsql, const char *userName, const char *password)
+{
+	PQExpBuffer command = NULL;
+	PGconn *connection = NULL;
+	char *escapedUsername = NULL;
+	char *escapedPassword = NULL;
+	bool success = true;
+
+	/* open a connection upfront since it is needed by PQescape functions */
+	connection = pgsql_open_connection(pgsql);
+	if (connection == NULL)
+	{
+		/* error message was logged in pgsql_open_connection */
+		return false;
+	}
+	escapedUsername = PQescapeIdentifier(connection, userName, strlen(userName));
+	if (escapedUsername == NULL)
+	{
+		log_error("Failed to set password for username \"%s\": %s", userName,
+				  PQerrorMessage(connection));
+
+		success = false;
+		goto cleanup;
+	}
+
+	escapedPassword = PQescapeLiteral(connection, password, strlen(password));
+	if (escapedPassword == NULL)
+	{
+		log_error("Failed to set password for username \"%s\": %s", userName,
+				  PQerrorMessage(connection));
+
+		success = false;
+		goto cleanup;
+	}
+
+	command = createPQExpBuffer();
+	appendPQExpBuffer(command, "ALTER USER %s", escapedUsername);
+	appendPQExpBuffer(command, " PASSWORD %s", escapedPassword);
```

Comment:
We would either use a PQExpBuffer and re-use the same escaped string memory area for both the username and the password, or use a string buffer of BUFSIZE characters and then use `sformat`. Given the code here, I think we should use the second approach.